### ISSUE TITLE
JsonConvert Default TypeNameHandling can break Xero API calls

### ISSUE THREAD

**Issue by J0nKn1ght** [XeroAPI/Xero-NetStandard#311]
**SDK:**
 - Version 3.13.1

**Describe the bug**
This is actually half bug and half feature suggestion.  If the application which is using Xero.NetStandard.OAuth2 has overridden the default JsonConvert settings, and changed the TypeNameHandling, it causes calls to the Xero API to fail with Validation Errors on the objects passed.

I don't know what is possible using the OpenApi tooling, but can additional JsonConvert settings be passed during the code generation?  If so, the settings should include `TypeNameHandling = TypeNameHandling.None`.

**To Reproduce**
(For example, in an MVC application Global.asax.cs file, or Startup.cs.)
Change the default JsonConvert settings:
```csharp
     JsonConvert.DefaultSettings = () => new JsonSerializerSettings
    {
        ContractResolver = new CamelCasePropertyNamesContractResolver(),
        TypeNameHandling = TypeNameHandling.Objects
    };
```

This results in JSON which includes the '$type' property for each object (e.g. for an invoice create operation):
```json
{
    "$type": "Xero.NetStandard.OAuth2.Model.Accounting.Invoices, Xero.NetStandard.OAuth2",
    "invoices": [
        {
            "$type": "Xero.NetStandard.OAuth2.Model.Accounting.Invoice, Xero.NetStandard.OAuth2",
            "type": "ACCREC",
            "lineAmountTypes": "Exclusive",
            "contact": {
                "$type": "Xero.NetStandard.OAuth2.Model.Accounting.Contact, Xero.NetStandard.OAuth2",
                "name": "Api Test"
            },
            "lineItems": [
                {
                    "$type": "Xero.NetStandard.OAuth2.Model.Accounting.LineItem, Xero.NetStandard.OAuth2",
                    "description": "Test Item",
                    "quantity": 1.0,
                    "unitAmount": 501.0,
                    "accountCode": "200"
                }
            ],
            "date": "2021-02-04T00:00:00Z",
            "dueDate": "2021-02-04T00:00:00Z"
        }
    ]
}
```

When the call is made, the response contains an invoice with the following validation errors:
- One or more line items must be specified
- Invoice Type must be specified
- A Contact must be specified for this type of transaction

**Expected behavior**
The default JsonConvert settings for the hosting application should not break the calls made by Xero.NetStandard.OAuth2.

**Comment by jamie-hoffmann-xero** [comment]
Hi @J0nKn1ght, thanks for reporting this issue. We've confirmed the bug and a fix is in progress. This will be implemented in the next release -- keep an eye out!

**Comment by jamie-hoffmann-xero** [comment]
Hi @J0nKn1ght, this problem should be addressed in version 3.14.2. Thanks again for reporting it, and please feel free to reopen the issue if you still have trouble.
